bloom-housing/bloom
# refactor: re-write header menu bar, remove bulma

# Pull Request Template

## Issue

Addresses #1345 #1467


## Description

We had been using Bulma as a dependency for our Site Header. We had four main issues with this, which were that customizing the header bar was becoming increasingly difficult as we had to override their styles and behavior, the Menu text was untranslated, some of the styles were just a little visually off, and the dropdown menu was not accessible on mobile.

This PR re-writes our SiteHeader from mostly scratch. There are new props to toggle certain visual choices on/off. I also refactored how we are sending in languages and the home page link to remove business logic. Also fixed a bug where header links were still showing in partners when signed-out.

## Type of change


## How Can This Be Tested/Reviewed?

Check out the new stories, however please note that Storybook is not accurately displaying the mobile view in its default viewing window. Would recommend starting locally or the deploy preview to view mobile.

In Bloom we are not flattening mobile menus, using the mobile menu button, using the mobile dropdown, and not showing the feedback bar on mobile.
In DAHLIA, they will be flattening mobile menus, using mobile text as a button, using the mobile drawer, and showing the feedback bar on mobile.

These are all toggleable features through props.

I would test:
* Accessibility in desktop and mobile (can you navigate through all things in the header and dropdowns using your keyboard, are they all focusable and actionable on enter)
* Changing the window sizes to view the transition between desktop and mobile
* Comparing visually this PR to https://dev-bloom.netlify.app/  and https://partners.bloom.exygy.dev/ - there are some styling differences which are purposeful based off of the designs here https://www.figma.com/file/tUtgsJO8XnF9Ij2dwu22EF/bloom-general?node-id=387%3A768

## Checklist:
